File: clients/python/SampleDaterClient.py
import socket
import time
import sys
from signal import signal, SIGPIPE, SIG_DFL  
signal(SIGPIPE,SIG_DFL)

clientsocket = socket.socket()


def generate_init_weight(N):
  '''
  TO DO: Change weights to your own algorithm
  DON'T FORGET THE \n
  '''

  weights = "1:-1" + ":0"*(N-2) + "\n"

  return weights


def generate_adjusted_weight(N, candidate_values):
  '''
  N: int
  candidates_values: string array
  '''
  
  '''
  TO DO: Change weights to your own algorithm
  DON'T FORGET THE \n
  '''
  
  weights = "1:-1" + ":0"*(N-2) + "\n"

  return weights
  

if __name__ == '__main__':
  clientsocket.connect(('localhost', 20000))
  socketfile = clientsocket.makefile('rw')
  socketfile.write('Person\n')
  socketfile.flush()
  inputLine = socketfile.readline()
  print("Server says:" + inputLine)
  N = (int) ((inputLine.split(":"))[1])
  if (N < 2 or N > 200):
    print("invalid N")
    sys.exit()
    
  # Generate the initial weights (First Round)
  time.sleep(.1)
  weights_init = generate_init_weight(N)
  socketfile.write(weights_init)
  socketfile.flush()
  
  # Generate the adjusted weights (Next Rounds)
  for i in range(20):
    inputLine = socketfile.readline()
    if inputLine:
      print("Server says:" + inputLine)
    candidate_values = inputLine.split(":")
    time.sleep(.1)
    weights = generate_adjusted_weight(N, candidate_values)
    socketfile.write(weights)
    socketfile.flush()
    
  clientsocket.close()

File: clients/java/SampleDaterClient.java
import java.io.*;
import java.net.*;
import java.util.*;

public class SampleDaterClient {

  public static void main(String[] args) throws Exception {
    Socket socket = null;
    PrintWriter out = null;
    BufferedReader in = null;
    try {
      socket = new Socket("localhost", 20000);
      out = new PrintWriter(socket.getOutputStream(), true);
      in = new BufferedReader(
        new InputStreamReader(socket.getInputStream()));
    } 
    catch (UnknownHostException e) {
      System.out.println("Unknown host");
      System.exit(-1);
    } 
    catch (IOException e) {
      System.out.println("No I/O");
      System.exit(-1);
    }  
    out.println("Person");
    String inputLine = in.readLine();
    System.out.println("Server says:" + inputLine);    
    int N = Integer.parseInt(inputLine.split(":")[1]);
    if(N < 2 || N > 200) {
      System.out.println("invalid N");
      System.exit(-1);
    }   
    Thread.sleep(100);
    String weights = generate_init_weight(N);
    out.println(weights);

    for(int p = 0; p < 20; p++) {
      inputLine = in.readLine();
      System.out.println("Server says:" + inputLine);
      String [] candidate_values = inputLine.split(":");
      Thread.sleep(100);    
      weights = generate_adjusted_weight(N, candidate_values);
      out.println(weights);
 
    }  
    out.close();
    in.close();
    socket.close();
  }
  private static String generate_init_weight(int N){
    // TO DO: Change weights to your own algorithm
    String weights ="1:-1";
    for(int i = 2; i < N; i++)
      weights += ":0";
    return weights;
  }

  private static String generate_adjusted_weight(int N, String [] candidate_values){
    // TO DO: Change weights to your own algorithm
    String weights ="1:-1";
    for(int i = 2; i < N; i++)
      weights += ":0";
    return weights;
  }
}